examples/lambda-pets/src/domain/model.ts

```diff
@@ -0,0 +1,44 @@
+import { Fx } from '../../../../src'
+
+// Domain model
+
+export type AdoptablePetsNear = {
+  readonly location: Location,
+  readonly radiusMiles: number,
+  readonly pets: Pets
+}
+
+export type Pets = readonly Pet[]
+
+export type Pet = {
+  readonly name: string,
+  readonly url: string,
+  readonly photoUrl?: string
+}
+
+export type GeoLocation = {
+  readonly longitude: number,
+  readonly latitude: number
+}
+
+export type Location = GeoLocation & {
+  readonly city?: string
+}
+
+export const defaultLocation: Location = {
+  latitude: 40.440624,
+  longitude: -79.995888,
+  city: 'Pittsburgh'
+}
+
+// Domain model access interfaces.
+// These are implemented by infrastructure
+
+export type GetPets<Effects> = (l: GeoLocation, radiusMiles: number) => Fx<Effects, Pets>
+
+export type GetLocation<Effects> = (host: string) => Fx<Effects, Location>
+
+export type PetsEnv<Effects> = {
```

Comment:
Perhaps this type belongs in application/pets.ts ?  The domain doesn't really need to know the property names, just the types.

Reply:
I feel like that makes sense to me to have them in application

Reply:
👍 That's my thinking as well, now that I've mulled it over a bit.